### PR TITLE
[CLEANUP] removed `Ember.required`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

**Comment by rwjblue** [comment]
I'm not 100% certain that this can be implemented in the addon. I believe we may need to go with the conditional flagging on this one.

@thoov - What do you think?

**Comment by thoov** [comment]
Agreed. Its going to have to be flagged. @bekzod I would look here for some examples: https://github.com/emberjs/ember.js/pull/15900

**Comment by rwjblue** [comment]
@bekzod - ping (when you have a chance)

**Comment by bekzod** [comment]
@rwjblue so it should be runtime flag ? flags are not stripped in build time ?

**Comment by rwjblue** [comment]
Confirm. It has to be a runtime flag because we cannot polyfill it from the ember-2-legacy addon.

**Comment by bekzod** [comment]
does this look better ?

File: packages/ember-metal/tests/mixin/required_test.js [hunk 1]
```diff
@@ -4,11 +4,14 @@ import {
...
   get
 } from '../..';
+import { ENV } from 'ember-environment';
 
 let PartialMixin, FinalMixin, obj;
-
+let originalEnvVal;
 QUnit.module('Module.required', {
   setup() {
+    originalEnvVal = ENV._ENABLE_PROPERTY_REQUIRED_SUPPORT
```
Comment (thoov) [inline packages/ember-metal/tests/mixin/required_test.js]: Missing a `;` here

File: packages/ember/lib/index.js [hunk 2]
```diff
@@ -127,7 +127,6 @@ Ember.observersFor = metal.observersFor;
 Ember.removeObserver = metal.removeObserver;
 Ember._suspendObserver = metal._suspendObserver;
 Ember._suspendObservers = metal._suspendObservers;
-Ember.required = metal.required;
```
Comment (thoov) [inline packages/ember/lib/index.js]: I believe we are going to have to guard this

File: packages/ember/tests/reexports_test.js [hunk 3]
```diff
@@ -108,7 +108,6 @@ QUnit.module('ember reexports');
   ['removeObserver', 'ember-metal'],
   ['_suspendObserver', 'ember-metal'],
   ['_suspendObservers', 'ember-metal'],
-  ['required', 'ember-metal'],
```
Comment (thoov) [inline packages/ember/tests/reexports_test.js]: I believe we are going to have to guard this

**Review by thoov — APPROVED** [review]

File: packages/ember/lib/index.js (hunk 2 above)
Comment (bekzod) [reply packages/ember/lib/index.js]: ok

File: packages/ember/tests/reexports_test.js (hunk 3 above)
Comment (bekzod) [reply packages/ember/tests/reexports_test.js]: ok

**Comment by thoov** [comment]
LGTM @rwjblue can you take a look

**Review by rwjblue — APPROVED** [review]

**Comment by krisselden** [comment]
In practice, this required doesn't do anything because of applyPartial and finishPartial doesn't handle it.  It only affects mixin.apply() which is rare.   For upgrading an addon could just make required() present and return null.

**Pull request merged**